src/ast/passes/semantic_analyser.cpp

```diff
@@ -126,6 +126,10 @@ void SemanticAnalyser::visit(Identifier &identifier)
     identifier.type = CreateInt(
         std::get<0>(getIntcasts().at(identifier.ident)));
   }
+  else if (func_ == "offsetof")
+  {
+    identifier.type = CreateNone();
+  }
```

Comment:
Why is this necessary?

Reply:
> Why is this necessary?

I am also curious, if there is no such sentence, it will report an error

```
./offsetof.bt:14:43-44: ERROR: Unknown identifier: 'x'
    printf("%ld\n", offsetof(struct Foo, x));
                                          ~
./offsetof.bt:16:51-56: ERROR: Unknown identifier: 'flags'
    printf("%ld\n", offsetof(struct task_struct, flags));
                                                  ~~~~~
./offsetof.bt:17:51-53: ERROR: Unknown identifier: 'mm'
    printf("%ld\n", offsetof(struct task_struct, mm));
                                                  ~~
./offsetof.bt:18:49-53: ERROR: Unknown identifier: 'next'
    printf("%ld\n", offsetof(struct list_head, next));
                                                ~~~~
./offsetof.bt:19:49-53: ERROR: Unknown identifier: 'prev'
    printf("%ld\n", offsetof(struct list_head, prev));
```

Reply:
Ah, makes sense, the field is an unknown identifier. I'm fine with keeping it as it is now, maybe just add a comment, something like:
```
// Handling the second arg of offsetof (a field name)
```